ssl/statem/statem_clnt.c

```diff
@@ -1688,7 +1689,8 @@ MSG_PROCESS_RETURN tls_process_server_certificate(SSL *s, PACKET *pkt)
     if ((SSL_IS_TLS13(s) && !PACKET_get_1(pkt, &context))
             || context != 0
             || !PACKET_get_net_3(pkt, &cert_list_len)
-            || PACKET_remaining(pkt) != cert_list_len) {
+            || PACKET_remaining(pkt) != cert_list_len
+            || PACKET_remaining(pkt) == 0) {
```

Comment:
I am not entirely convinced that the spec 100% requires certificate_list to be nonempty.  It is possible, however unlikely, that some future certificate_type code point is defined that feels a need to send the Certificate message but with no actual content.
That said, I think it's better to have this check than not have this check, as it is a good safety check to have for the currently supported certificate types.